postgres/src/main/scala/package.scala

```diff
@@ -0,0 +1,127 @@
+package anorm
+
+import java.util.UUID
+
+import scala.util.control.NonFatal
+
+import play.api.libs.json.{
+  JsError, JsObject, Json, JsSuccess, JsValue, Reads, Writes }
+
+import org.postgresql.util.PGobject
+
+package object postgresql extends PGJson {
+
+  /** Instance of `ToSql` to support PostgreSQL UUID parameter */
+  implicit val UUIDToSql: ToSql[UUID] = ToSql[UUID] { _ => "?::UUID" -> 1 }
+}
+
+// Could be moved to a separate module
+sealed trait PGJson {
+  /** Allows to pass a `JsValue` as parameter to be stored as `PGobject`. */
+  implicit def jsValueToStatement[J <: JsValue] = ToStatement[J] { (s, i, js) =>
+    val pgObject = new PGobject()
+    pgObject.setType(JsValueParameterMetaData.sqlType)
+    pgObject.setValue(Json.stringify(js))
+    s.setObject(i, pgObject, JsValueParameterMetaData.jdbcType)
+  }
+
+  /**
+    * @tparam the type of value to be written as JSON
+    * @param value the value to be passed as a JSON parameter
+    * @param w the Play writes to be used to serialized the value as JSON
+    * 
+    * {{{
+    * import play.api.libs.json._
+    * import anorm._, postgresql._
+    * 
+    * case class Foo(bar: String)
+    * implicit val w: Writes[Foo] = Json.writes[Foo]
+    * 
+    * val value = Foo("lorem")
+    * SQL"INSERT INTO test(id, json) VALUES(${"bar"}, ${value})".executeUpdate()
+    * }}}
+    */
+  def asJson[T](value: T)(implicit w: Writes[T]): ParameterValue = {
+    implicit val writeJsonToStatement: ToStatement[T] =
+      jsValueToStatement[JsValue].contramap(w.writes)
+
+    ParameterValue from value
+  }
+
+  implicit object JsValueParameterMetaData extends ParameterMetaData[JsValue] {
+    val sqlType = "JSONB"
+    val jdbcType = java.sql.Types.OTHER
+  }
+
+  implicit object JsObjectParameterMetaData
+      extends ParameterMetaData[JsObject] {
+    val sqlType = "JSONB"
+    val jdbcType = java.sql.Types.OTHER
+  }
+
+  /**
+    * {{{
+    * import play.api.libs.json.JsValue
+    * import anorm.{ SQL, SqlParser }
+    * import anorm.postgresql._
+    * 
+    * SQL"SELECT json FROM test".as(SqlParser.scalar[JsValue].single)
+    * }}}
+    */
+  implicit val jsValueColumn: Column[JsValue] =
+    Column.nonNull[JsValue] { (value, meta) =>
+      @inline def str: Option[String] = value match {
+        case o: PGobject => Some(o.getValue)
+        case s: String => Some(s)
+        case clob: java.sql.Clob => Some(
+          clob.getSubString(1, clob.length.toInt))
+      }
+
+      str match {
```

Comment:
What I mean is that `str` only generates `Some` values. In other words you could probably change `str` to

```scala
@inline def str: String = value match {
  case o: PGobject => o.getValue)
  case s: String => s
  case clob: java.sql.Clob =>
  clob.getSubString(1, clob.length.toInt)
}
```